src/utils/errors.ts

```diff
@@ -0,0 +1,196 @@
+/**
+ * A generic Error that all other JOSE specific Error subclasses extend.
+ *
+ * @example Checking thrown error is a JOSE one
+ *
+ * ```js
+ * if (err instanceof jose.errors.JOSEError) {
+ *   // ...
+ * }
+ * ```
+ */
+export class JOSEError extends Error {
+  /** A unique error code for the particular error subclass. */
+  static get code(): string {
+    return 'ERR_JOSE_GENERIC';
+  }
+
+  /** A unique error code for the particular error subclass. */
+  code: string = 'ERR_JOSE_GENERIC';
+
+  constructor(message?: string) {
+    super(message);
+    this.name = this.constructor.name;
+    // @ts-ignore
```

Comment:
Is this required?

Reply:
It's useful if you want to throw a generic error. For now we don't have this case but I preferred to insert it anyway to make it the same as the others!